libraries/standard/mqtt/src/mqtt.c

```diff
@@ -99,6 +101,427 @@ static int32_t sendPacket( MQTTContext_t * pContext,
+{
+    MQTTStatus_t status = MQTTSuccess;
+    int32_t bytesReceived = 0;
+    int32_t totalBytesReceived = 0;
+    size_t bytesToReceive = 0;
+
+    assert( pContext != NULL );
+
+    if( incomingPacket.remainingLength > pContext->networkBuffer.size )
+    {
+        bytesToReceive = pContext->networkBuffer.size;
+        IotLogErrorWithArgs( "Incoming packet length %u exceeds network buffer size %u."
+                             "Incoming packet will be dumped",
+                             incomingPacket.remainingLength
+                             pContext->networkBuffer );
+    }
+    else
+    {
+        bytesToReceive = incomingPacket.remainingLength;
+    }
+
+    bytesReceived = pContext->transportInterface.recv( pContext->transportInterface.networkContext,
+                                                       pContext->networkBuffer.pBuffer,
+                                                       bytesToReceive );
+    totalBytesReceived = bytesReceived;
+
+    /* Partial receive. */
+    if( bytesReceived < ( int32_t ) bytesToReceive )
+    {
+        IotLogWarnWithArgs( "Packet partially received: Bytes received=%d,"
+                            "Bytes expected=%u",
+                            bytesReceived,
+                            bytesToReceive );
+        /* Handle partial receive. */
+        bytesReceived = pContext->transportInterface.recv( pContext->transportInterface.networkContext,
+                                                           ( pContext->networkBuffer.pBuffer + totalBytesReceived ),
+                                                           ( bytesToReceive - totalBytesReceived ) );
+        totalBytesReceived += bytesReceived;
+        if( totalBytesReceived != ( int32_t ) bytesToReceive )
+        {
+            IotLogErrorWithArgs( "Handling partial receive failed. Bytes "
+                                 "received=%d, Bytes expected=%u",
+                                 totalBytesReceived,
+                                 bytesToReceive );
+            status = MQTTRecvFailed;
+        }
+    }
+    /* Too many bytes received. */
+    else if( bytesReceived > ( int32_t ) bytesToReceive )
+    {
+        IotLogErrorWithArgs( "Too many bytes received. Bytes received=%d, ",
+                             "Bytes expected=%u",
+                             bytesReceived,
+                             bytesToReceive );
+        status = MQTTRecvFailed;
+    }
+    else
+    {
+        /* Receive successful, bytesReceived == bytesToReceive. */
+        IotLogInfoWithArgs( "Packet received. Received bytes=%d",
+                            bytesReceived );
+    }
+
+    /* Check if packet exceeds buffer. */
+    if( ( status == MQTTSuccess ) && ( incomingPacket.remainingLength > bytesToReceive ) )
+    {
+        /* Packet exceeds buffer, dump it. */
+        status = dumpPacket( pContext, incomingPacket.remainingLength, totalBytesReceived );
+    }
+
```

Comment:
An empty-else branch is needed for MISRA 15.7 compliance.

Reply:
Why? There aren't any `else if`s

Reply:
Oh, nevermind then